torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -318,32 +398,68 @@ def should_pad_mm(match: Match) -> bool:
     )
 
 
-def mm_replace(mat1: Tensor, mat2: Tensor) -> Tensor:
-    m_padded_length = get_padded_length(mat1.shape[0], get_alignment_size(mat1))
-    k_padded_length = get_padded_length(mat1.shape[1], get_alignment_size(mat1))
-    n_padded_length = get_padded_length(mat2.shape[1], get_alignment_size(mat2))
-
-    return pad_mm(mat1, mat2, m_padded_length, k_padded_length, n_padded_length)
-
-
 def pad_mm(
     mat1: Tensor,
     mat2: Tensor,
     m_padded_length: int,
     k_padded_length: int,
     n_padded_length: int,
+    explicit_transpose: bool = False,
 ) -> Tensor:
-    # mm_replace will go through pad_mm multiple times if multiple dimensions are needed to be padded
-    if k_padded_length != 0:
-        mat1 = pad_dim(mat1, k_padded_length, 1)
-        mat2 = pad_dim(mat2, k_padded_length, 0)
-        return torch.ops.aten.mm(mat1, mat2)
-    elif n_padded_length != 0:
-        mat2 = pad_dim(mat2, n_padded_length, 1)
-        return torch.ops.aten.mm(mat1, mat2)[:, :-n_padded_length]
+    if k_padded_length != 0 or m_padded_length != 0:
+        # dim order is reversed for constant_pad_nd, for every dim we specify right and left padding
+        mat1_padded = aten.constant_pad_nd(
+            mat1, [0, k_padded_length, 0, m_padded_length]
+        )
+    else:
+        mat1_padded = mat1
```

Comment:
Nit: this if else could get refactored into a helper and reused

Reply:
Just checked, while there are three very similar code fragments, they have important differences that are not easily abstracted in a helper function without making it harder to read. So I'll leave this as is.